### ISSUE TITLE
Question about elasticsearch attachments and jdbc river

### ISSUE THREAD

**Issue by istvano** [jprante/elasticsearch-jdbc#286]
Hi Guys,

I am using elasticsearch to index my db table in mysql. I am using the river to fetch data from mysql and send it to ES. 

The additional complexity is the following:

I have a filepath column in mysql. This is a reference to a binary file ( word document, pdf ) in the filesystem which is actually part of the data and I need to send that to ES as an attachement as well. As I need the content of the file indexed as well.

Could someone help me to figure out what are the right strategies to do this?

Should I implement a custom strategy?
Should I move all my files to DB instead of filesystem ? is ES jdbc river able to send the file to ES.  Would this approach slow down indexing considerably ?

I would appreciate any suggestions or ways I should go with.

Thanks in advance,
Istvan

**Comment by ilkermutlu** [comment]
@istvano , did you come up with a solution to this? I need to implement something very similar.

**Comment by istvano** [comment]
Hi,
I ended up using this -> 
https://github.com/searchisko/elasticsearch-river-remote

I have built a tiny API which is called byt this remote river and handles everything. deletes, updates, incremental and full. 

I think JDBC river is great but this is a special case which probably easier to deal with outside of it.
I am not sure if this helps you but this is what I came up with.

Kind Regards,

**Comment by jprante** [comment]
@istvano do you use JDBC with https://github.com/searchisko/elasticsearch-river-remote and if so, how do you recognize deletes?

**Comment by istvano** [comment]
Hi, I am using the remote api at the moment as I can load the files from file system. 
Basically the solution is that in the mapping I can set up a field and a value. if that field contains the configured value for a given row the river treats it as a delete instead of update.

Support for data deletes

River supports correct update of search indices for two basic types of data deletes in remote system.

If deleted data simply disappear from the remote system API responses then they are deleted from search index at the end of next full update. It is not possible to catch this type of deletes during incremental updates.

If deleted data are marked by some flag only and correctly timestamped to be returned by your system in next incremental update request, then you can use remote_field_deleted and remote_field_deleted_value river config params to point river to this flag and delete data from search index even during incremental update. Configured delete flag is reflected during full update also. This feature is available from 1.6.2 version of the river.

Note: You have to correctly set analyzers for some fields in mapping to allow correct deletes from search index, see previous chapter!

**Comment by jprante** [comment]
@istvano yes, I know how RESTful APIs can recognize deletes because there are document identifiers (the URI), but I guess that the solution you found is not using JDBC result sets where there are no identifiers for documents?

I just ask because I would like to improve JDBC plugin by a document identifier mechanism. Maybe with the help of an SQL result set to URI mapper.
